include/net/rest_client.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file rest_client.h
+ *
+ * @defgroup rest_client REST client library
+ * @{
+ * @brief REST client.
+ *
+ * @details Provide REST client functionality.
+ */
+
+#ifndef REST_CLIENT_H__
+#define REST_CLIENT_H__
+
+#include <net/http_parser.h>
+
+/** @brief TLS is not used. */
+#define REST_CLIENT_NO_SEC -1
+
+/** @brief Use default TLS peer verification, i.e., TLS_PEER_VERIFY_REQUIRED. */
+#define REST_CLIENT_TLS_DEFAULT_PEER_VERIFY -1
+
+/** @brief REST client does a sckt connection */
+#define REST_CLIENT_SCKT_CONNECT -1
+
+/** @brief Common HTTP status codes */
+enum rest_client_http_status {
+	REST_CLIENT_HTTP_STATUS_OK = 200,
+	REST_CLIENT_HTTP_STATUS_BAD_REQ = 400,
+	REST_CLIENT_HTTP_STATUS_UNAUTH = 401,
+	REST_CLIENT_HTTP_STATUS_FORBIDDEN = 403,
+	REST_CLIENT_HTTP_STATUS_NOT_FOUND = 404,
+};
+
+/**
+ * @brief REST client context.
+ *
+ * @details Parameters and data for using the REST client API.
+ *          Request related parameters should be filled before calling @ref rest_client_request
```

Comment:
Would it make sense to group the `req`/`rsp` part into two substructures within `struct rest_client_req_resp_context` or is it too much hassle?

Reply:
I was also thinking that during implementation time. But I think that I was starting to think is it ok if for example:  
`resp_buff` would be in `req `and `response ` ptr pointing to that from separate `resp`.